src/Events/Telemetry/Telemetry.php

```diff
@@ -228,14 +230,20 @@ public function inject_modal_link() {
 			return;
 		}
 
+		// 'the-events-calendar'
 		$telemetry_slug = \TEC\Common\Telemetry\Telemetry::get_plugin_slug();
 
+		$show = get_option( Config::get_container()->get( Opt_In_Template::class )->get_option_name( $telemetry_slug ) );
+
+		if ( ! $show ) {
+			return;
+		}
 		/**
 		 * Fires to trigger the modal content on admin pages.
 		 *
 		 * @since TBD
 		 */
-		do_action( "stellarwp/telemetry/{$telemetry_slug}/optin" );
+		do_action( 'tec-telemetry-modal', $telemetry_slug );
```

Comment:
why not `tec_telemetry_modal_slug`?

Reply:
We trigger `stellarwp/telemetry/{$telemetry_slug}/optin` in common, but rely on the plugins themselves to tell us _where_ they want it - ie in the TEC settings pages

Could change it to `tec_telemetry_modal` though (still no slug - it's meant to be generic